socket_windows.go

```diff
@@ -0,0 +1,181 @@
+package kr
```

Comment:
I wasn't very comfortable with the amount of cut-n-paste either, but I thought it might make you more confident in merging if the pull request didn't change socket.go :-)   ... so yes.  I'll take a shot at refactoring to make sure everything common is in socket.go